actors/builtin/miner/miner_actor.go

```diff
@@ -2104,6 +2126,9 @@ func processEarlyTerminations(rt Runtime) (more bool) {
 	// TODO: We're using the current power+epoch reward. Technically, we
 	// should use the power/reward at the time of termination.
 	// https://github.com/filecoin-project/specs-actors/v6/pull/648
+	// TODO #2: for the fix to issue 799 we precompute these in OnEpochTickEnd and
```

Comment:
This is a great observation and is a good next step.  The place correct method to extend is `OnDeferredCronEvent`.  Both the `processEarlyTermination` and `handleProvingDeadline` subfunctions redundantly call the same reward and power actor queries.  Please file an issue for this and remove this TODO from code.  While some legacy TODOs like this one exist I have been trying to remove them in place of GH issues.

Reply:
ok, opened the issue: #1486